### DISCUSSION TITLE
Question

### DISCUSSION THREAD

**Discussion by Astell-Chen** [geodynamics/sw4#158] [Q&A] [answered]
Hi,
Does SW4 support set a cavity in the centre of model? 

Thanks

**Comment by andersp** [comment, marked as answer]
Short answer is no, but you can model the cavity as a light material (low density). In that case the wave speeds are unchanged, but the particle displacement/velocity will be very large inside the cavity. This will probably lead to unphysically large reflections, but could be worth a try. Some experimentation will be required to get reasonable results.

**Reply by kazmanza** [reply]
I've modeled cavities (underground excavations) before just by setting low density and very low velocities. I've not tested or investigated the results too closely in SW4, but have done it numerous times in e3d, and it seemed to work. If there really is a massive, stark transition from solid ground to open air, large reflections are expected. To make it more realistic, one can add a transition zone around the cavity where the velocities gradually decrease (probably more realistic due to damage/fracture zone that will exist around any underground cavity).

**Reply by Astell-Chen** [reply]
Thank your reply very much! Is E3D an open source code? If so, how can I get it?

**Reply by andersp** [reply]
E3d is not open source. This forum is intended for the SW4 code :-)
